sdks/java/io/amqp/src/main/java/org/apache/beam/sdk/io/amqp/AmqpIO.java

```diff
@@ -157,26 +149,21 @@ public Read withMaxNumRecords(long maxNumRecords) {
      * {@link PCollection}.
      */
     public Read withMaxReadTime(Duration maxReadTime) {
-      checkArgument(maxNumRecords() == Long.MAX_VALUE,
-          "maxNumRecord and maxReadTime are exclusive");
```

Comment:
They can, but I think it's better to have a consistent rule (validate the property itself in the builder method, validate cross-property invariants in expand)